tasks/eva_2121/aggregate_table.py

```diff
@@ -0,0 +1,226 @@
+    if reports and len(reports) == 1:
+        report = reports[0]
+
+    if fasta is None or report is None:
+        report, fasta = download_assembly(scientific_name, assembly_accession, download_dir)
+    return fasta, report
+
+
+def download_assembly(scientific_name, assembly_accession, download_dir):
+    python = os.path.join(eva_accession_path, "python-import-automation/bin/python")
+    genome_downloader = os.path.join(eva_accession_path, "eva-accession/eva-accession-import-automation/genome_downloader.py")
+    private_json = os.path.join(eva_accession_path, "private-config.json")
+
+    output_dir = os.path.join(download_dir, scientific_name.lower().replace(' ', '_'))
+    os.makedirs(output_dir, exist_ok=True)
+    command = "{} {} -p {} -a {} -o {}""".format(python, genome_downloader, private_json, assembly_accession, output_dir)
+    run_command_with_output('Retrieve assembly fasta and assembly report', command)
+
+    assembly_report = glob.glob(os.path.join(output_dir, assembly_accession, "*_assembly_report.txt"))[0]
+    assembly_fasta = os.path.join(output_dir, assembly_accession, assembly_accession + ".fa")
+    return assembly_fasta, assembly_report
+
+
+def count_variants(rows, only_variant_to_process=True):
+    """Sum variants in the rows provided."""
+    stay_unchanged = {row['Will stay unchanged'] for row in rows}
+    return sum([
+        int(row['Number Of Variants (submitted variants)'].replace(',', ''))
+        for row in rows
+        if not only_variant_to_process or ('no' in stay_unchanged and row['Source'] != 'DBSNP - filesystem')
+    ])
+
+
+def aggregate_list_of_species(input_file, assembly_dir, download_dir, output_assemblies_tsv, output_taxonmomy_tsv):
+    """
+    """
+    data_per_taxid, data_per_taxid_and_assembly = parse_input(input_file)
+    only_unmapped_tax_id = []
+    unchanged_taxid = []
+    dbsnps_only = []
+    eva_only = []
+    mixture = []
+    taxid_to_tempmongo = {}
+    total_number_variant = 0
+    for taxid in data_per_taxid:
+        rows = data_per_taxid[taxid]
+        sources = {row['Source'] for row in rows}
+        stay_unchanged = {row['Will stay unchanged'] for row in rows}
+
+        number_variants = count_variants(rows)
+        total_number_variant += number_variants
+
+        source = None
+        if len(sources) == 1:
+            source = sources.pop()
+        if source == 'DBSNP - filesystem':
+            only_unmapped_tax_id.append(taxid)
+        elif 'no' not in stay_unchanged:
+            unchanged_taxid.append(taxid)
+        elif source == 'EVA':
+            eva_only.append(taxid)
+        elif source == 'DBSNP':
+            dbsnps_only.append(taxid)
+        else:
+            mixture.append(taxid)
+
+    # Iterate a second time to assign a temp mongodb per taxonomy
+    with open(output_taxonmomy_tsv, 'w') as open_output:
+        headers = ['taxonomy_id', 'scientific_name', 'temp_mongo', 'to_copy', 'number_variants_to_process', 'number_variants']
```

Comment:
Suggest ```headers = ['taxonomy_id', 'scientific_name', 'tempmongo_instance', 'should_be_copied', 'num_variants_in_taxid', 'total_num_variants']```

Reply:
Changed excepted for `number_variants_to_process` which I prefer over `num_variants_in_taxid`

Reply:
Also, could we include the dbsnp name since we use that in the release folder name?

Reply:
That a good point to bring up. we should discuss the naming convention currently used.